doc/README.txt

```diff
@@ -1,11 +1,45 @@
 maptlotlib documentation
 ========================
 
+
+Building the documentation
+--------------------------
+
+A list of dependencies can be found in ../doc-requirements.txt.
+
+All of these dependencies can be installed through pip::
+
+  pip install -r ../doc-requirements.txt
+
+or conda::
+
+  conda install sphinx numpydoc ipython mock colorspacious pillow
+
+To build the HTML documentation, type ``python make.py html`` in this
+directory. The top file of the results will be ./build/html/index.html
+
+**Note that Sphinx uses the installed version of the package to build the
+documentation**: matplotlib must be installed *before* the docs can be
+generated. Even if that is the case, one of the files needed to do this,
+'../lib/matplotlib/mpl-data/matplotlibrc', is not version controlled, but
+created when matplotlib is built. This means that the documentation cannot be
+generated immediately after checking out the source code, even if matplotlib
+is installed on your system: you will have to run ``python setup.py build``
+first.
```

Comment:
The problem is that some examples try to fetch data on internet and the website is down.
I don't think we should be have download from internet required to build the documentation.
I personally am in favor of removing all the examples that rely on downloading data on internet.

It turns out that all of these examples are from the finance module, which is deprecated.

(Thanks so much to @rougier for the help in debugging).